Library/Homebrew/extend/os/linux/keg_relocate.rb

```diff
@@ -37,21 +36,21 @@ def change_rpath(file, old_prefix, new_prefix)
     lib_path = "#{new_prefix}/lib"
     rpath << lib_path unless rpath.include? lib_path
     new_rpath = rpath.join(":")
-    cmd = [patchelf, "--force-rpath", "--set-rpath", new_rpath]
 
     if file.with_interpreter?
-      old_interpreter = Utils.safe_popen_read(patchelf, "--print-interpreter", file).strip
+      old_interpreter = patchelf.interpreter
       new_interpreter = if File.readable? "#{new_prefix}/lib/ld.so"
         "#{new_prefix}/lib/ld.so"
       else
         old_interpreter.sub old_prefix, new_prefix
       end
-      cmd << "--set-interpreter" << new_interpreter if old_interpreter != new_interpreter
     end
 
     return if old_rpath == new_rpath && old_interpreter == new_interpreter
 
-    safe_system(*cmd, file)
+    patchelf.runpath = new_rpath
+    patchelf.interpreter = new_interpreter
+    patchelf.save
```

Comment:
What if this fails? Will it raise an exception?

Reply:
There's no official documentation, glancing over the source code, it won't throw an exception. 
Based on the assumption that the file is executable,
rpath, interpreter section exists( checked above ),
 and we have write permission to directory.

Reply:
Thanks!